Enhanced sat list for M8N/M9N

TonyBlit opened the pull request:
Modified the sat list in order to show extra info from the following PR: https://github.com/betaflight/betaflight/pull/10921

New layout:
![Screenshot 2021-09-17 at 06 36 43](https://user-images.githubusercontent.com/2266058/133726563-20a95fc0-a58c-49b1-8b81-7d800c383b87.png)

New sat list in action:
<img width="403" alt="Screenshot 2021-09-09 at 17 42 34" src="https://user-images.githubusercontent.com/2266058/133726700-95873ade-889a-4722-bb0e-3c8a2c6efe6b.png">

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

haslinghuis commented on src/tabs/gps.html:
```diff
@@ -45,104 +45,218 @@
-                                <td style="width: 20%;" i18n="gpsSignalSatId"></td>
-                                <td style="width: 15%;" i18n="gpsSignalQty"></td>
-                                <td style="width: 65%;" i18n="gpsSignalStr"></td>
+                                <!--<td style="width: 10%;" i18n="gpsSignalGnssId"></td>-->
```
Should this be removed?

TonyBlit replied:
Actually, I forgot to ask about this. How does the localization system work? do I have to add gpsSignalGnssId with the "Gnss Id" string to the messages.json in all languages? In this case it's easy, but what about other strings that might need to be translated?

McGiverGim replied:
Not, only English. We have translators that will do the rest of languages

haslinghuis commented on src/js/tabs/gps.js:
```diff
@@ -64,12 +64,109 @@ TABS.gps.initialize = function (callback) {
+                            $('td', row).eq(0).text('-');
+                    }
+
+                    if (FC.GPS_DATA.chn[i] >= 7) {
```
```suggestion
                    if (FC.GPS_DATA.chn[i] >== 7) {
```

haslinghuis commented on src/js/tabs/gps.js:
```diff
@@ -64,12 +64,109 @@ TABS.gps.initialize = function (callback) {
+                        let quality = '';
+                        switch (FC.GPS_DATA.quality[i] & 0x7) {
+                            case 0:
+                                quality = quality + 'no signal | ';
```
```suggestion
                                quality += 'no signal | ';
```

TonyBlit replied:
I have reworked those ugly switch cases, so this is no longer an issue :-)

haslinghuis reviewed (changes requested):
Thanks for contributing.
Please remove commented out code.
Some remarks I made are valid in different places.

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> FAIL
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

haslinghuis commented on src/tabs/gps.html:
```diff
@@ -45,104 +45,216 @@
+                                <td>0</td>
                             </tr>
-                        </table>
+                            </table>
```
👾 was on the right position 😆

TonyBlit replied:
ouch :-)

haslinghuis reviewed (commented):
Please also fix the sonar and squash your commits when done. Looks good.

McGiverGim commented:
> Please also fix the sonar and squash your commits when done. Looks good.

Only as a help for @TonyBlit , not all the Sonar issues need to be fixed. Sometimes they have no sense. In this case, the literal that has been duplicated I think it does not need to be fixed, it is part of an array and will be strange to define this outside and not the other elements. 
The other two issues can be fixed.

TonyBlit commented:
@McGiverGim yeah, the first one it doesn't make sense, indeed. The other ones made me think I forgot to localize the messages, so I've fixed everything and squashed, hope all is good now.

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

McGiverGim reviewed (approved):
To me is good enough... good work! :)
As suggestion, in the Configurator PRs is good to attach a screen capture. It helps with the review.

haslinghuis reviewed (dismissed):
LGTM

McGiverGim commented:
@TonyBlit I've seen this in the firmware PR:

> gps_ublox_use_galileo will no longer disable QZSS as it's suggested on the M8N manual

Can you modify this description:

https://github.com/betaflight/betaflight-configurator/blob/dfd8a925d04042c52a8786789e8673a14a455c08/locales/en/messages.json#L1465-L1468

I think something like:

> It enables the Galileo system (European). Until Betaflight 4.2 this implied to remove the QZSS system (Japanese) too.

Sorry, I'm not English native, I'm sure you will find something better :)

TonyBlit commented:
@McGiverGim Well spotted! I was not even conscious of this message.

I've been tempted to mention ublox_use_galileo is only for M8N, but given M9N has galileo already enabled, maybe it's not worth complicating the sentence until M9N is widely adopted.

limonspb reviewed (changes requested):
@TonyBlit please squash in 1 commit  and i think its ready to go.

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> PASS

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

chmelevskij commented on src/js/tabs/gps.js:
```diff
@@ -64,12 +69,57 @@ TABS.gps.initialize = function (callback) {
+            } else {
+                // M8N/M9N on newer firmware
+
+                let channels = FC.GPS_DATA.chn.length;
+                if (channels > 32) {
+                    channels = 32; //the list in html can only show 32 channels but future firmware could send more
+                }
```
```suggestion
                const maxUIChannels = 32;
                let channels = Math.min(maxUIChannels, FC.GPS_DATA.chn.length);
```

TonyBlit replied:
good suggestion, thanks!

Asizon commented on src/tabs/gps.html:
```diff
@@ -45,104 +45,216 @@
-                                <td style="width: 20%;" i18n="gpsSignalSatId"></td>
-                                <td style="width: 15%;" i18n="gpsSignalQty"></td>
-                                <td style="width: 65%;" i18n="gpsSignalStr"></td>
+                                <td style="width: 12%;" i18n="gpsSignalGnssId"></td>
```
The use of styles into html part is old, now we try to move all styles in to css file.

chmelevskij replied:
I wouldn't block this though. Both inline styles and styles files are redundant if we just move to vue, so one way or the other re-work will be needed.

Asizon reviewed (changes requested):
Really good features, my requested changes are not to important but if you can take a look would be nice

haslinghuis commented:
Sonar disabled us here from merging this 😕
